abce/trade.pyx

```diff
@@ -544,8 +543,7 @@ class Trade:
                 raise NotEnoughGoods(self.name, offer.good, quantity - available)
             if quantity > available:
                 quantity = available
-            self._haves[offer.good] -= quantity
-            self._haves['money'] += quantity * offer.price
+            self._haves.transform(offer.good, offer.price, 'money', quantity)
```

Comment:
It is a transformation from good into money, since money is defined as physical good in the library.